slack/plugin.py

```diff
@@ -17,24 +17,38 @@ class SlackUser:
     display_name: str
 
 
-def _with_mentions(
-    users: list[SlackUser] | Callable[[], list[SlackUser]],
+async def _with_mentions(
+    users: Union[
+        list[SlackUser],
+        Callable[[], list[SlackUser]],
+        Callable[[], Coroutine[Any, Any, list[SlackUser]]],
+    ],
     message: str,
 ) -> str:
     """
     Return message with @-mentions using display names replaced by Slack IDs.
 
     Args:
-        users: The list of SlackUsers to consider.  This can also be a function that
-            returns the list.  It is only called if there are any @-mentions in message.
+        users: The list of SlackUsers to consider. This can also be a function that
+            returns the list, or an async function that returns the list.
```

Comment:
```suggestion
        users: The list of SlackUsers to consider.  This can also be an awaitable that
            will only be awaited if there are any @-mentions in the message.
```